packages/rest-specs/specs/authentication/http/custom/mockapi.ts

```diff
@@ -0,0 +1,58 @@
+import { MockRequest, passOnSuccess, ScenarioMockApi } from "@typespec/spec-api";
+import { getValidAndInvalidScenarios } from "../../commonapi.js";
+
+export const Scenarios: Record<string, ScenarioMockApi> = {};
+
+const validAndInvalidScenarios = getValidAndInvalidScenarios(
+  "http/custom",
+  "invalid-api-key",
+  function addOptionalParamOldApiVersionNewClientValidate(req: MockRequest): void {
+    req.expect.containsHeader("authorization", "SharedAccessKey valid-key");
+  },
+);
+
+Scenarios.Authentication_Http_Custom_Valid_Key = passOnSuccess({
+  uri: `/authentication/http/custom/valid`,
+  mockMethods: [
+    {
+      method: "get",
+      request: {
+        config: {
+          headers: {
+            authorization: "SharedAccessKey valid-key",
+          },
+        },
+      },
+      response: {
+        status: 204,
+      },
+    },
+  ],
+});
+
+Scenarios.Authentication_Http_Custom_InValid_Key = passOnSuccess({
+  uri: `/authentication/http/custom/invalid`,
+  mockMethods: [
```

Comment:
feel like it would have been great to get this PR with only one spec out first we'd review the new API as now this is going to be a pain to change..

There is a few issues I feel with this:
1. A lot of uncessary nesting for basic cases
2. Won't allow calling different urls in a row(e.g. paging)

I think this should:
1. Either take a single `MockApiDefinition` or an array of `ModelApiDefinition[]`
2. Do we need `config`, why are headers or other props not directly at the root of the request
3. In response `data` is not obvious as where that data is. Should this be named body and depend on `json()`

Reply:
1. The Logic of generating is changed. Even the paging scenario is included in the new model.
2. I am taking a MockApiDefinition
3. As for config, this is useful since we could easily handle the server call.

Reply:
As mentioned in the other one config feels like it's tied to axios which is an implementation detail.

How is that more helpful?

Reply:
`mockMethods` is a list because there can be multiple methods for the same endpoint. They can all be defined in the same place. I agree that the original idea of config was taken from axios. But, it makes sense to keep all config related properties under one object instead of multiple properties. Don't you think?

Reply:
As mentioned in another comment, config is removed and its properties are moved up.